packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -2,7 +2,7 @@
 from _plotly_utils.basevalidators import ColorscaleValidator
 from ._core import apply_default_cascade, init_figure, configure_animation_controls
 from .imshow_utils import rescale_intensity, _integer_ranges, _integer_types
-import pandas as pd
+import narwhals.stable.v1 as nw
```

Comment:
Would the `numpy` import here need to be moved like you did with `def ols...`?

Reply:
Yes it could be moved, but at the end it is imported already in `.imshow_utils` anyway, so this import should be free.
I didn't want to go through the headache to figure out if numpy could be made optional in `imshow_utils` for now 😇

Reply:
Thanks! Is `numpy` required even if `pandas` isn't? 
I tried importing `plotly.express` and I don't have `numpy` installed, so it fails because `_imshow` is imported here
https://github.com/plotly/plotly.py/pull/4790/files#diff-483af8f56c64a3426bdc9b5c597e218f407a16ca311fa1db1841d2a2ddbcf7f5L14

Reply:
That's great news, it seems definitly possible to make numpy optional as well then! I would gladly give a try as follow up

Reply:
@FBruzzesi Could you re-add the error message removed in `packages/python/plotly/plotly/express/__init__.py` but edit it to refer to `numpy` rather than `pandas`?

Ideally `numpy` should be made optional altogether since it seems to be only required in a few places, but until then we should at least provide a helpful error message.